src/armonik_cli/commands/sessions.py

```diff
@@ -0,0 +1,206 @@
+    type=str,
+    default="default",
+    show_default=True,
+    help="Default partition.",
+    metavar="PARTITION"
+)
+@click.option(
+    "--application-name",
+    type=str,
+    required=False,
+    help="Default application name.",
+    metavar="NAME"
+)
+@click.option(
+    "--application-version",
+    type=str,
+    required=False,
+    help="Default application version.",
+    metavar="VERSION"
+)
+@click.option(
+    "--application-namespace",
+    type=str,
+    required=False,
+    help="Default application namespace.",
+    metavar="NAMESPACE"
+)
+@click.option(
+    "--application-service",
+    type=str,
+    required=False,
+    help="Default application service.",
+    metavar="SERVICE"
+)
+@click.option(
+    "--engine-type",
+    type=str,
+    required=False,
+    help="Default engine type.",
+    metavar="ENGINE_TYPE"
+)
+@click.option(
+    "--option",
+    type=str,
+    required=False,
+    multiple=True,
+    help="Additional default options.",
+    metavar="NAME=VALUE"
+)
+@output_option
+@debug_option
+@errors.error_handler
+def create(endpoint: str, max_retries: int, max_duration: str, priority: int, partition: List[str], default_partition: str, application_name: str | None, application_version: str | None, application_namespace: str | None, application_service: str | None, engine_type: str | None, option: List[str] | None, output: str, debug: bool):
+    """Create a new session."""
+
+    # Validate max_duration format.
+    try:
+        max_duration = _parse_time_delta(max_duration)
+    except ValueError:
+        raise click.BadParameter("Invalid format for max duration. Use HH:MM:SS.MS.")
+
+    # Validate options format.
+    options = {}
+    for item in option:
+        try:
+            k, v = item.split("=")
+            options[k] = v
+        except ValueError:
+            raise click.BadParameter(f"Invalid format for option: {item}. Use key=value.")
+
+    with grpc.insecure_channel(endpoint) as channel:
+        sessions_client = ArmoniKSessions(channel)
+        session_id = sessions_client.create_session(
+            default_task_options=TaskOptions(
+                max_duration=max_duration,
+                priority=priority,
+                max_retries=max_retries,
+                partition_id=default_partition,
+                application_name=application_name,
+                application_version=application_version,
+                application_namespace=application_namespace,
+                application_service=application_service,
+                engine_type=engine_type,
+                options=options
+            ),
+            partition_ids=partition
+        )
+    return console.get_console().formatted_print({"SessionId": session_id}, format=output)
+
+
+def _clean_up_status(session: Session) -> Session:
+    session.status = SessionStatus.name_from_value(session.status).split("_")[-1].capitalize()
+    return session
+
+
+def _parse_time_delta(time_str: str) -> timedelta:
```

Comment:
I think that this is provided in armonik api

Reply:
There is no such function within the API. However, there are similar functions to convert from gRPC objects to Python objects.